presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcPreparedStatement.java

```diff
@@ -166,6 +168,67 @@ public void testGetMetadata()
         }
     }
 
+    @Test
+    public void testGetParameterMetaData()
+            throws Exception
+    {
+        try (Connection connection = createConnection("blackhole", "blackhole")) {
+            try (Statement statement = connection.createStatement()) {
+                statement.execute("CREATE TABLE test_get_parameterMetaData (" +
+                        "c_boolean boolean, " +
+                        "c_decimal decimal, " +
+                        "c_decimal_2 decimal(10,3)," +
+                        "c_varchar varchar, " +
+                        "c_row row(x integer, y array(integer)), " +
+                        "c_array array(integer), " +
+                        "c_map map(integer, integer))");
+            }
+
+            try (PreparedStatement statement = connection.prepareStatement(
+                    "SELECT ? FROM test_get_parameterMetaData WHERE c_boolean = ? AND c_decimal = ? " +
+                            "AND c_decimal_2 = ? AND c_varchar = ? AND c_row = ? AND c_array = ? AND c_map = ?")) {
+                ParameterMetaData parameterMetaData = statement.getParameterMetaData();
+                assertEquals(parameterMetaData.getParameterCount(), 8);
+
+                assertEquals(parameterMetaData.getParameterClassName(1), "unknown");
+                assertEquals(parameterMetaData.getParameterType(1), Types.NULL);
+                assertEquals(parameterMetaData.getParameterTypeName(1), "unknown");
+
+                assertEquals(parameterMetaData.getParameterClassName(2), Boolean.class.getName());
+                assertEquals(parameterMetaData.getParameterType(2), Types.BOOLEAN);
+                assertEquals(parameterMetaData.getParameterTypeName(2), "boolean");
+
+                assertEquals(parameterMetaData.getParameterClassName(3), BigDecimal.class.getName());
+                assertEquals(parameterMetaData.getParameterType(3), Types.DECIMAL);
+                assertEquals(parameterMetaData.getParameterTypeName(3), "decimal");
```

Comment:
It would be better if we add test for scale and precision

Reply:
I tested the cases for `decimal(10,3)` and `varchar(5)` as the rest values are determined by default, thus I don't think they need to be tested.